src/script/components/manifest-options.ts

```diff
@@ -426,6 +463,10 @@ export class AppManifest extends LitElement {
     this.manifest.theme_color = (<HTMLInputElement>event.target).value;
   }
 
+  handleModalInputFileChange() {
+    console.log('handleModalInputFileChange');
+  }
+
   addNewScreenshot() {
     this.screenshotList.push(undefined);
```

Comment:
So if you change this to something like ```this.screenshotList = [...this.screenshotList, undefined] ``` which creates a new instance of the array you should be able to get rid of the requestUpdate in the line below. It is my understanding that the reason this works this way is that lit-element does not do "deep" change detection and therefore does not see that array length change. However, when you set it to a new instance of that array it will trigger a change.

Reply:
oh good thinking!